liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/AbstractLibertySupport.java

```diff
@@ -309,20 +296,23 @@ protected ArtifactItem createArtifactItem(String groupId, String artifactId, Str
 
             if (resolvedDependencies.isEmpty()) {
                 // No matching artifacts were found in the resolved dependencies. Send warning.
-                log.warn("Unable to find artifact matching coordinates " + gavCoordinates + " in either project dependencies or in project dependencyManagement.");
+                log.warn("Unable to find artifact matching groupId "+ groupId +", artifactId "+artifactId+", and version "+version+" in either project dependencies or in project dependencyManagement.");
             }
         }
 
         return resolvedDependencies;
      }
 
      protected void findTransitiveDependencies(Artifact resolvedArtifact, Set<Artifact> resolvedArtifacts, Set<Artifact> resolvedDependencies) {
+        boolean isProvidedScopeAllowed = resolvedArtifact.getScope().equals(Artifact.SCOPE_PROVIDED);
         String coords = resolvedArtifact.getGroupId() + ":" + resolvedArtifact.getArtifactId() + ":";
         for (Artifact artifact : resolvedArtifacts) {
-            // Do not copy transitive dependencies with SCOPE_PROVIDED
-            if (!artifact.getScope().equals(Artifact.SCOPE_PROVIDED)) {
+            // Do not copy transitive dependencies with SCOPE_PROVIDED unless the resolvedArtifact is SCOPE_PROVIDED.
```

Comment:
I thought we were trying to do more like: "don't copy transitive dependencies of a provided scope dependency.. only copy the original, top-level dependency".

Reply:
This is what we discussed yesterday, where if the top level dependency is provided scope, then its compile/runtime dependencies show up as provided scope. They should still get copied. This code is necessary to allow things like the derbyshared dependency to get copied. If you reference the same dependency as compile scope, then its transitive dependency shows up as its original scope.

![image (2)](https://user-images.githubusercontent.com/29436451/89683152-a1037b80-d8bd-11ea-8317-f3858f87a2b9.png)